### PR TITLE
fixes issue where build de-dup doesn't add project analyzers.

### PR DESCRIPTION
**Pull request by heejaechang**
fix https://github.com/dotnet/roslyn/issues/11378

### CODE REVIEW

**Comment by heejaechang** [comment]
test vsi please

**Comment by heejaechang** [comment]
test vsi please

File: src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateManager.cs [hunk 1]
```diff
@@ -122,19 +122,27 @@ public void RemoveStateSet(ProjectId projectId)
             /// </summary>
             public ImmutableArray<StateSet> CreateBuildOnlyProjectStateSet(Project project)
             {
+                // create project analyzer reference identity map
                 var referenceIdentities = project.AnalyzerReferences.Select(r => _analyzerManager.GetAnalyzerReferenceIdentity(r)).ToSet();
```
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateManager.cs]: v1 and v2 has same implementation. we will remove v1 once update 3 ship.

File: src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs [hunk 2]
```diff
@@ -113,19 +113,27 @@ public StateSet GetOrCreateStateSet(Project project, DiagnosticAnalyzer analyzer
...
                 var compilerAnalyzer = _analyzerManager.GetCompilerDiagnosticAnalyzer(project.Language);
                 StateSet compilerStateSet;
-                if (stateSetMap.TryGetValue(compilerAnalyzer, out compilerStateSet))
+                if (hostStateSetMap.TryGetValue(compilerAnalyzer, out compilerStateSet))
                 {
                     stateSets.Add(compilerStateSet);
                 }
 
+                // now add all project analyzers
+                stateSets.AddRange(_projectStates.GetOrUpdateStateSets(project));
```
Comment (heejaechang) [inline src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs]: this was missing.

**Comment by heejaechang** [comment]
@mavasani @davkean @srivatsn @dotnet/roslyn-analysis can you take a look?

**Comment by mavasani** [comment]
:+1:

**Pull request merged**